native/logikSimulation/src/data/subnet.rs

```diff
@@ -0,0 +1,20 @@
+/// Representing a subnet with a certain state
+pub(crate) struct Subnet {
+    state: SubnetState,
+}
+
+impl Subnet {
+    pub(crate) fn new() -> Self {
+        Self {
+            state: SubnetState::Floating,
+        }
+    }
+}
+
+/// An enum to represent the different states that a subnet can have
+pub(crate) enum SubnetState {
```

Comment:
When implementing logic regarding states I would recommend having a look at [IEEE 1364 Chapter 5.1.10](https://www.eg.bucknell.edu/~csci320/2016-fall/wp-content/uploads/2015/08/verilog-std-1364-2005.pdf) to see how the bitwise operations should work.
I've made an efficient C# implementation that can be used as a reference [here](https://github.com/inda19plusplus/logik/blob/master/src/LogikUI/Simulation/Value.cs).